connector/profilingmetricsconnector/connector.go

```diff
@@ -54,6 +66,25 @@ func (c *profilesToMetricsConnector) ConsumeProfiles(ctx context.Context, profil
 }
 
 func (c *profilesToMetricsConnector) Start(ctx context.Context, host component.Host) error {
+	c.aggregations = make([]aggregation, 0, len(c.config.CustomAggregations))
+
+	for _, agg := range c.config.CustomAggregations {
+		if agg.Label == "" || agg.Match == "" {
+			// Both, label and match, need to be set.
+			return errInvalAgg
+		}
+
+		// Precompile all regular expressions at startup to do it only once.
+		re, err := regexp.Compile(agg.Match)
+		if err != nil {
+			return err
```

Comment:
Maybe we can return a better error here that gives more context:

```suggestion
			return fmt.Errorf("compiling rx for label %v: %v", agg.Label, err)
```